### PR TITLE
Add a Pull Request template

### PR DESCRIPTION
**Pull request by adarsh**
Reason for Change
=================
* Consistent Pull Requests are good because they allow us to communicate in a unified way.
* [GitHub offers a convention for this kind of template via their `.github/` folder.][1]

Changes
=======
* Add a PR template in the "RFC" or "Request for Change" format.

[1]: https://github.com/blog/2111-issue-and-pull-request-templates

### CODE REVIEW

**Comment by adarsh** [comment]
This might be too extensive, but I wanted to start the discussion about unifying the PR format at least.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+=========
+_Put an `x` in the boxes that apply.
+You can also fill these out after creating the PR._
+
+- [ ] I have linked to all relevant reference issues or work requests
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] I have added or updated necessary documentation (if appropriate)
+- [ ] Any dependent changes have been merged and published in downstream
+
+Recommended Reviewers
```
Comment (jessieay) [inline .github/PULL_REQUEST_TEMPLATE.md]: with the "request review" feature, I don't find this section to be necessary

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+Minor
+=====
+* List any minor changes or style fixes here.
+
+Risks
+=====
+* Describe any risks you see to existing systems this fix or feature might have
+if those systems where to incorporate this change.
+
+Checklist
```
Comment (jessieay) [inline .github/PULL_REQUEST_TEMPLATE.md]: I would probably not vote to have this section - I do like the "checklist manifesto"-ness of it, but I find it best to have automated things makes these types of checks so I don't have to do the manual work of checking boxes or expecting others to do so (yay overcommit / Circle / code climate!)

Comment (adarsh) [inline .github/PULL_REQUEST_TEMPLATE.md]: I agree - removed in d267f63

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (adarsh) [inline .github/PULL_REQUEST_TEMPLATE.md]: Excellent point - removed in d267f63

**Comment by adarsh** [comment]
> guessing we should wait on @jayroh 👀 before merging?

Probably, but he doesn't have GH access yet so I'm going to merge and ask for his edits after the fact.

Helps a ton that we are opening/closing PRs at such a pace!

**Pull request merged**